### PR TITLE
Update CHANGELOG to describe ROCm 5.7 contribution

### PR DESCRIPTION
**Pull request by causten**
Changes are based on internal review of all the PRs since 5.6.  Should not contain incomplete features, or inter-release fixes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,33 +2,50 @@
...
 ### Added
-- Y-Model feature to store tuning information with the optimized model
-- Added Python 3.10 bindings 
-- Accuracy checker tool based on ONNX Runtime
-- ONNX Operators parse_split, and Trilu 
-- Build support for ROCm MLIR
-- Added migraphx-driver flag to print optimizations in python (--python)
-- Added JIT implementation of the Gather and Pad operator which results in better handling of larger tensor sizes.
-
+- Enabled hipRTC allowing the ROCm install to not be required in /opt/rocm specifically
```
Comment (pfultz2) [inline CHANGELOG.md:8]: We still support rocm in a different location than /opt/rocm without hipRTC. This should be: `Enabled hipRTC to not require dev packages for migraphx runtime and allow the ROCm install to be in a different directory than it was during build time`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,33 +2,50 @@
...
-- Accuracy checker tool based on ONNX Runtime
-- ONNX Operators parse_split, and Trilu 
-- Build support for ROCm MLIR
-- Added migraphx-driver flag to print optimizations in python (--python)
-- Added JIT implementation of the Gather and Pad operator which results in better handling of larger tensor sizes.
-
+- Enabled hipRTC allowing the ROCm install to not be required in /opt/rocm specifically 
+- Add support for multi-target execution
+- Added Dynamic Batch support with C++/Python APIs
+- Enabled native int32 type support
```
[the remark this answers is not in the document]
Comment (causten) [reply CHANGELOG.md:11]: Nice catch it came from https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/1721.  Reading the PR message I think it should be removed

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,33 +2,50 @@
...
+- Fix assertion error during verify and make DCE work with tuples
+- Fix convert operation for NaNs
+- Fix shape typo in API test
+- Fix compile warnings for shadowing variable names
+- Add missing specialization for the `nullptr` for the hash function
 
 ### Changed
-- Changed version/location of 3rd party build dependencies to pick up fixes
+- Bumped version of half library to 5.6.0
+- Use clang-format to format
```
Comment (pfultz2) [inline CHANGELOG.md:43]: We have always used clang-format to format. This should be `Use /opt/rocm/llvm/bin/clang-format to format`.

File: CHANGELOG.md (hunk 1 above)
Comment (causten) [reply CHANGELOG.md:8]: ok

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,33 +2,50 @@
...
+- Bumped CI to support rocm 5.6
+- Make building tests optional
+- Update install prereqs python fix
```
Comment (pfultz2) [inline CHANGELOG.md:46]: What does this mean? Its not even using english grammar

File: CHANGELOG.md (hunk 3 above)
Comment (causten) [reply CHANGELOG.md:43]: That would assume there is an /opt/rocm link as opposed to what hipRTC is suppose to do

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,33 +2,50 @@
 
 Full documentation for MIGraphX is available at [MIGraphX Documentation](https://rocmdocs.amd.com/projects/AMDMIGraphX/en/latest/).
 
-## MIGraphX 2.5 for ROCm 5.5.0
+## MIGraphX 2.7 for ROCm 5.7.0
```
Comment (pfultz2) [inline CHANGELOG.md:5]: I would assume a CHANGELOG would be appended not replaced for each version. Like how rocm-cmake does the CHANGELOG:
https://github.com/RadeonOpenCompute/rocm-cmake/blob/develop/CHANGELOG.md

File: CHANGELOG.md (hunk 3 above)
Comment (pfultz2) [reply CHANGELOG.md:43]: > That would assume there is an /opt/rocm link

There is always an /opt/rocm link as we added in our dockerfile.

> as opposed to what hipRTC is suppose to do

Formatting is done at dev time not at runtime.

File: CHANGELOG.md (hunk 5 above)
Comment (causten) [reply CHANGELOG.md:5]: Yup makes sense.

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,8 +2,49 @@
...
+- Load specific version of the migraphx_py library
+- Annotate functions with the block size so the compiler can do a better job of optimizing 
+- Enable reshape on nonstandard shapes
+- Use half HIP APIs to compute max and min
+- Added support for broadcasted scalars to unsqueeze operator
+- Improved multiplies with dot operator
+- Handle broadcasts across dot and concat
+- Add verify namespace for better symbol resolution
+### Fixed
+- Resolved accuracy issues with resnet50
```
Comment (umangyadav) [inline CHANGELOG.md:30]:
```suggestion
- Resolved accuracy issues with FP16 resnet50
```

**Review by CharlieL7 — APPROVED** [review]

**Pull request merged**